### PR TITLE
Updating printing

### PR DESCRIPTION
**Pull request by john-westcott-iv**
Changing from `print` to `click.echo`.
Adding printWarning method (prints magenta) and printError method (prints red) to distinguish messages. 
Implemented our own pprint method to use click.echo w/ json.dumps.
Added a few try/catch blocks to grab some errors found during testing.
Added two spaces to fix column headers in status command.

### CODE REVIEW

File: receptorctl/receptorctl/cli.py [hunk 1]
```diff
@@ -28,6 +27,17 @@ def parse_args(self, ctx, args):
                 param.required = False
             return super(IgnoreRequiredWithHelp, self).parse_args(ctx, args)
 
+def pprint(json_data):
+    click.echo(json.dumps(json_data, indent=4, sort_keys=True))
+
+def printWarning(message):
```
Comment (shanemcd) [inline receptorctl/receptorctl/cli.py]: Python function names should be `snake_case`. https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

File: receptorctl/receptorctl/cli.py [hunk 2]
```diff
@@ -401,10 +409,10 @@ def cancel(ctx, unit_ids):
 @click.pass_context
 def release(ctx, force, unit_ids):
     if len(unit_ids) == 0:
-        print("No unit IDs supplied: Not doing anything")
+        printWarning("No unit IDs supplied: Not doing anything")
         return
     op = "release" if not force else "force-release"
-    print("Released:")
+    click.echo("Released:")
```
Comment (shanemcd) [inline receptorctl/receptorctl/cli.py]: I like that you added helper functions for errors and warnings, but dislike that we `click.echo` repeated everywhere. I don't care what you call the new function (`print_info`?), but we should apply a similar pattern here to future-proof this code a bit.

File: receptorctl/receptorctl/cli.py [hunk 3]
```diff
@@ -28,6 +27,17 @@ def parse_args(self, ctx, args):
...
 
+def pprint(json_data):
+    click.echo(json.dumps(json_data, indent=4, sort_keys=True))
+
+def printWarning(message):
+    click.echo(click.style(f"Warning: {message}", fg='magenta'), err=True)
+
+def printError(message, exitCode=None):
+    click.echo(click.style(f"ERROR: {message}", fg='red'), err=True)
+    if exitCode:
```
Comment (shanemcd) [inline receptorctl/receptorctl/cli.py]: I don't like the idea of exiting the program as a side effect of a print function, it makes the code difficult to follow. I had to go through the rest of your diff and find when you were passing `1` as an (unnamed) argument to this function.

**Review by shanemcd — CHANGES_REQUESTED** [review]

File: receptorctl/receptorctl/cli.py (hunk 1 above)
Comment (john-westcott-iv) [reply receptorctl/receptorctl/cli.py]: Fixed.

File: receptorctl/receptorctl/cli.py (hunk 2 above)
Comment (john-westcott-iv) [reply receptorctl/receptorctl/cli.py]: Initially I only added error and warning to make sure we got a consistent color and prefix. But added a generic print_message function as well.

File: receptorctl/receptorctl/cli.py (hunk 3 above)
Comment (john-westcott-iv) [reply receptorctl/receptorctl/cli.py]: I probably should have made a print_error_and_exit function to be more explicit but went back to just a print_error followed by sys.exit.

**Comment by shanemcd** [comment]
The test failure here is strange. I don't see how what you did caused that, but we aren't seeing it on other runs. I don't have time to investigate today, curious if you can figure out what's going on.

**Comment by shanemcd** [comment]
Got curious and tried this locally. I was able to reproduce, so should be easy to track down when you have time. Just make sure you have `tox` installed and run `make receptorctl-test`.

**Comment by john-westcott-iv** [comment]
@shanemcd I found the issue. There were two print commands in the format `print("string", variable)`. I missed them when converting to print_message so they were not properly displaying their variables which were parsed in the tests.

**Review by shanemcd — APPROVED** [review]

**Pull request merged**